demo-app/src/tx_hooks_impl.rs

```diff
@@ -64,11 +66,46 @@ impl<C: Context> TxHooks for DemoAppTxHooks<C> {
             // therefore this panic should never happen, we add it for sanity check.
             .unwrap_or_else(|e| panic!("Inconsistent nonce {e}"));
     }
+
+    fn enter_apply_batch(
+        &self,
+        sequencer: &[u8],
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> Result<()> {
+        // TODO: Handle errors
+        match self.sequencer_hooks.next_sequencer(working_set) {
+            Ok(next_sequencer) => {
+                if next_sequencer != sequencer {
+                    // TODO: Return an error, should we slash in this case?
+                    todo!()
```

Comment:
I don't think we can slash here, because the invalid sequencer almost definitely won't happen to be bonded. 

Plus, this mechanism could potentially get abused to get other sequencers slashed (by holding onto a stale bundle that they signed and then posting it later. So I think we just want to skip execution

Reply:
Done: Only error is returned.